### PR TITLE
MINOR: use StoreBuilder in KStreamImpl rather than StateStoreSupplier

### PR DESCRIPTION
**Pull request by dguy**

### CODE REVIEW

**Comment by dguy** [comment]
@guozhangwang, @bbejeck, @mjsax

File: streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java [hunk 1]
```diff
@@ -827,16 +828,16 @@ public boolean test(final K1 key, final V1 value) {
...
-            .withValues(valueSerde)
-            .persistent()
-            .windowed(windows.size(), windows.maintainMs(), windows.segments, true)
-            .build();
+    private static <K, V> StoreBuilder<WindowStore<K, V>> createWindowedStateStore(final JoinWindows windows,
+                                                                                   final Serde<K> keySerde,
+                                                                                   final Serde<V> valueSerde,
+                                                                                   final String storeName) {
+        return Stores.windowStoreBuilder(Stores.persistentWindowStore(storeName,
+                                     windows.maintainMs(),
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java]: nit fix indention

**Review by mjsax — APPROVED** [review]
One nit. Otherwise LGTM

**Review by bbejeck — APPROVED** [review]
Thanks @dguy LGTM

**Comment by guozhangwang** [comment]
LGTM.

**Comment by dguy** [comment]
merged to trunk

**Pull request closed without merging**